docs/Dealing-with-Duplication-of-Data.md

```diff
@@ -0,0 +1,108 @@
+# Internal consistency of system descriptions
+
+There are cases where different parts of the system description describe the same aspect of a system in different ways. When a system description is generated by an inspection of an existing system, all parts of the description are consistent. But when modifying a description it can happen that inconsistencies are created by modifying different parts of the description in different ways. This document describes our approach how to handle the internal consistency of system descriptions.
+
+
+## Examples of inconsistencies
+
+When a system is inspected some data is duplicated. This happens for example when multiple scopes handle and store the same information.
+An example is repositories: The repositories are queried and then stored in `manifest.json` by the repositories scope.
+Additionally the files in `/etc/zypp/repos.d` contain the same information and are handled by unmanaged-files.
+Another example are users and groups, which are handled by the scopes 'users' and 'groups'. The files `/etc/passwd`,
+`/etc/shadow`, and `/etc/group` contain the same information. Configuration files are handled by the scope config-files.
+
+Duplication of this data leads to problems like the same repository appearing multiple times on a build image because it's applied by the unmanaged-files and the repositories scope.
+
+Another example are mount points. They are present in the description as the files where this information is contained. For changing this information or using it from other tools it would be desirable to have this information available in a more abstract form as well.
+
+
+## Current Approach
+
+The current approach is to avoid inconsistencies by filtering files duplicating information which is stored in the manifest. This is done either at inspection or on export.
+
+During inspection for some scopes (e.g. for users and groups) all the information contained in the configuration files (`/etc/passwd` and `/etc/group` in the example) is put into the manifest. The file where the information is coming from then is not extracted by the inspection.
```

Comment:
We should keep this section updated in the future to reflect the current implementation of filters (custom filtes, predefined filters).